lib/private.h

```diff
@@ -67,6 +67,11 @@ typedef struct {
 
 struct migration;
 
+struct vfu_sparse_mmap_areas {
+    int nr_mmap_areas;
+    struct vfu_mmap_area areas[];
```

Comment:
Shall we replace `struct vfu_mmap_area` with `struct iovec`. The latter exists exactly for that reason and we don't have to introduce yet another data type.

Reply:
iovec.iov_base is (void *) , I think instead of user passing address, it is better to just pass value the way we now have looks convenient. I am fine either way.

Reply:
I don't think the type of `iovec.iov_base` is of a substantial significance here.